tests/helpers/stubs.py

```diff
@@ -335,6 +349,44 @@ def isActive(self):
         return self._started
 
 
+class FakeConfigSection:
+
+    """A stub for a KeyValue entry in configdata.DATA."""
+
+    def __init__(self, *entries):
+        self.values = collections.OrderedDict()
+        self.descriptions = {}
+        for name, value, desc in entries:
+            self.values[name] = value
+            self.descriptions[name] = desc
+
+    def __iter__(self):
+        """Iterate over all set values."""
+        return self.values.__iter__()
+
+    def __getitem__(self, key):
+        return self.values[key]
+
+
+class FakeSettingValue:
```

Comment:
> Though I guess this makes it easier to customize valid_values and default for tests?

Exactly. If I use a real `SettingValue`, then I have to use a real `ConfigType`, and my test becomes flaky with regards to changes in configtypes (e.g. if someone added an extra option to `IgnoreCase`).